static/js/components/NewGroupUserForm.jsx

```diff
@@ -46,54 +37,19 @@ const NewGroupUserForm = ({ group_id }) => {
     }
   }, [dispatch, allUsers]);
 
-  const submitAndResetForm = async () => {
-    let result = null;
-    if (formState.invitingNewUser && invitationsEnabled) {
-      result = await dispatch(
-        invitationsActions.inviteUser({
-          userEmail: formState.newUserEmail,
-          groupIDs: [group_id],
-          groupAdmin: [formState.admin],
-          streamIDs: null,
-        })
-      );
-      if (result.status === "success") {
-        dispatch(
-          showNotification(
-            `Invitation successfully sent to ${formState.newUserEmail}`
-          )
-        );
-        setFormState({
-          newUserEmail: null,
-          admin: false,
-          invitingNewUser: false,
-        });
-      }
-    } else {
-      result = await dispatch(
-        groupsActions.addGroupUser({
-          username: formState.newUserEmail,
-          admin: formState.admin,
-          group_id,
-        })
-      );
-      if (result.status === "success") {
-        setFormState({
-          newUserEmail: null,
-          admin: false,
-          invitingNewUser: false,
-        });
-      }
-    }
-  };
-
-  const handleClickSubmit = (event) => {
-    event.preventDefault();
-    if (invitationsEnabled && formState.invitingNewUser) {
-      // If user clicks confirm, `submitAndResetForm` will be called
-      setConfirmDialogOpen(true);
-    } else {
-      submitAndResetForm();
+  const handleClickSubmit = async () => {
```

Comment:
I think it would good to add a validation check in the frontend form here that checks for username !== null. Otherwise, the error is caught thrown from the backhand and the end user sees a "UserID is missing" error which may be confusing since the form refers to usernames.

Reply:
:+1: Done